### PR TITLE
bug(cart): fix bug where clearing an empty cart failed

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
Attempting to clear an empty cart results in an error.

Fixes: N/A

## What is the new behavior?
Fixed.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/cart/src/drivers/magento/cart.service.ts [hunk 1]
```diff
@@ -53,14 +53,17 @@ export class DaffMagentoCartService implements DaffCartServiceInterface<DaffCart
...
+        items.length ? zip(...items.map(item =>
           this.cartItemDriver.delete(cartId, item.item_id)
-        ))
+        )) : of(null)
       ),
       // find the most up to date delete response by looking for one with no items
-      // it might be necessary to make a get request instead if delete does not return needed fields
-      map(updatedCarts => updatedCarts.find(cart => cart.items.length === 0))
+      // make a get request if there were no items in the cart
+			switchMap(updatedCarts => !!updatedCarts ?
```
Comment (griest024) [inline libs/cart/src/drivers/magento/cart.service.ts]: I think `!!` is unnecessary if you don't actually need an explicit boolean. For the purposes of a conditional it should be sufficient to do `updatedCarts ?`.

**Comment by lderrickable** [comment]
> What do you think about doing this logic in the effects layer instead? It would save an additional `get` request. Also, if this is behavior that we want across daffodil as a whole, it would be nice not to have to ensure consistent behavior across multiple drivers.
> 
> LMK your thoughts on this.

No, it should be here, I think, because other platforms might have a clear cart mutation rather than just a remove item which would make it unnecessary. We should treat this as a magento-only problem, I think

File: libs/cart/src/drivers/magento/cart.service.ts (hunk 1 above)
Comment (lderrickable) [reply libs/cart/src/drivers/magento/cart.service.ts]: changed

**Comment by griest024** [comment]
@lderrickable good point

**Pull request merged**